addons/mail/static/src/new/utils.js

```diff
@@ -154,3 +154,32 @@ export function useFocus(refName, callback = () => {}) {
     );
     return state;
 }
+
+export function useVisible(refName) {
+    const ref = useRef(refName);
+    const state = useState({ isVisible: false });
+    const observer = new IntersectionObserver((entries) => {
+        for (const entry of entries) {
+            state.isVisible = entry.isIntersecting;
+            console.log(entry.isIntersecting);
+        }
+    });
+    let el;
+    onMounted(observe);
+    onWillUnmount(observe);
+    onPatched(observe);
+
+    function observe() {
+        if (ref.el !== el) {
+            if (el) {
+                observer.unobserve(el);
+            }
+            if (ref.el) {
+                observer.observe(ref.el);
+            }
+        }
+        el = ref.el;
+    }
```

Comment:
Are you sure the method is properly covering the `onWillUnmount` case?
If the component is still in its normal state when the hook is called, `ref.el` will be equal `el` and it will fail to `unobserve`.

Reply:
Change done (ensure call to `observer.unobserve()` in `onWillUnmount`